Utilities/Administration/github_publish_os_package_list.py

```diff
@@ -0,0 +1,85 @@
+# specific language governing permissions and limitations
+# under the License.
+
+import jwt
+import sys
+import time
+import base64
+import distro
+import argparse
+import requests
+
+argparser = argparse.ArgumentParser()
+argparser.add_argument('--private_key', help='File path the to private key used for JWT signing')
+argparser.add_argument('--app_id', help='GitHub App ID')
+argparser.add_argument('--installation_id', help='GitHub App installation ID')
+argparser.add_argument('--repository', help='OWNER/NAME of the GitHub repository (eg. data-team-uhn/cards)')
+argparser.add_argument('--deployment_hostname', help='Directory name of the host (eg. cardsdemo.uhndata.io)')
+args = argparser.parse_args()
+
+THIS_DISTRO = distro.id()
+
+with open(args.private_key, 'r') as f:
+	PRIVATE_KEY = f.read()
+
+def updateFile(path, new_content_file, installation_token):
+	installation_headers = {}
+	installation_headers['Accept'] = 'application/vnd.github+json'
+	installation_headers['Authorization'] = 'token ' + installation_token
+
+	prev_version_resp = requests.get("https://api.github.com/repos/" + args.repository + "/contents/" + path, headers=installation_headers)
+	prev_version_sha = prev_version_resp.json()['sha']
+
+	commit = {}
+	commit['message'] = "Updating " + path
+	commit['committer'] = {'name': 'update_os_package_list.py', 'email': 'user@localhost'}
+	with open(new_content_file, 'rb') as f:
+		commit['content'] = base64.b64encode(f.read()).decode()
+	commit['sha'] = prev_version_sha
+
+	r = requests.put("https://api.github.com/repos/" + args.repository + "/contents/" + path, headers=installation_headers, json=commit)
+	if r.status_code not in range(200, 300):
+		raise Exception("Failed to update " + path)
+
+payload = {}
+payload['iat'] = int(time.time())
+payload['exp'] = payload['iat'] + (60 * 10)
+payload['iss'] = args.app_id
+
+github_jwt = jwt.encode(payload, PRIVATE_KEY, algorithm='RS256')
+
+application_headers = {}
+application_headers['Accept'] = 'application/vnd.github+json'
+application_headers['Authorization'] = 'Bearer ' + github_jwt.decode()
+
+installation_token_resp = requests.post("https://api.github.com/app/installations/" + args.installation_id + "/access_tokens", headers=application_headers)
+installation_token = installation_token_resp.json()['token']
+
+if THIS_DISTRO == 'debian':
+	updateFile(args.deployment_hostname + "/vm/os-release", "/usr/lib/os-release", installation_token)
+	updateFile(args.deployment_hostname + "/vm/debian_version", "/etc/debian_version", installation_token)
+	updateFile(args.deployment_hostname + "/vm/status", "/var/lib/dpkg/status", installation_token)
+elif THIS_DISTRO == 'ubuntu':
+	updateFile(args.deployment_hostname + "/vm/os-release", "/usr/lib/os-release", installation_token)
+	updateFile(args.deployment_hostname + "/vm/lsb-release", "/etc/lsb-release", installation_token)
+	updateFile(args.deployment_hostname + "/vm/status", "/var/lib/dpkg/status", installation_token)
```

Comment:
Would it make sense to do this instead:

```suggestion
if THIS_DISTRO == 'debian' || THIS_DISTRO == 'ubuntu':
	updateFile(args.deployment_hostname + "/vm/os-release", "/usr/lib/os-release", installation_token)
	updateFile(args.deployment_hostname + "/vm/os-version", "/etc/debian_version" if THIS_DISTRO == 'debian' else "/etc/lsb-release", installation_token)
	updateFile(args.deployment_hostname + "/vm/status", "/var/lib/dpkg/status", installation_token)
```

There's less duplicated code, and there's only one file on our end holding the version.

Reply:
I've addressed this code duplication issue in commit https://github.com/data-team-uhn/cards/pull/1158/commits/6a5114999ccaae25daec070455394c302d48cd09 by adding the `os_file_management.py` simple module. I don't want to use the generic `/vm/os-version` solution that you have proposed as, in doing so, we loose the information of what the original filename was and thus when we need to recreate the simple file structure in a test Docker image (CARDS-1896) we will not know what `/vm/os-version` should be renamed to - `/etc/debian_version` or `/etc/lsb-release`?

Reply:
OK, sounds good.